cranelift-codegen/src/regalloc/register_set.rs

```diff
@@ -261,6 +287,50 @@ mod tests {
         classes: &[],
     };
 
+    const RSI_1: RegSetIter = RegSetIter {
+        regs: [0x31415927, 0x27182818, 0x14141356],
+    };
+
+    const RSI_2: RegSetIter = RegSetIter {
+        regs: [0x00000000, 0x00000000, 0x00000000],
+    };
+
+    const RSI_3: RegSetIter = RegSetIter {
+        regs: [0xffffffff, 0xffffffff, 0xffffffff],
+    };
+
+    fn reverse_regset_iteration_work(rsi: &RegSetIter) {
+        // Check the reverse iterator by comparing its output with the forward iterator.
+        let mut rsi_f = (*rsi).clone();
+        let mut results_f = Vec::<RegUnit>::new();
+        while let Some(r) = rsi_f.next() {
+            results_f.push(r);
+        }
+
+        let mut rsi_r = (*rsi).clone();
+        let mut results_r = Vec::<RegUnit>::new();
+        while let Some(r) = rsi_r.rnext() {
+            results_r.push(r);
+        }
+
+        let len_f = results_f.len();
+        let len_r = results_r.len();
+        assert_eq!(len_f, len_r);
+
+        if len_f == len_r {
```

Comment:
This check is unneeded, the test would assert (above) and panic before this is reached.